torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -939,15 +939,33 @@ void ProcessGroupNCCL::enableCollectivesTiming() {
   enableTiming_.store(true);
 }
 
-c10::optional<std::thread> ProcessGroupNCCL::tryWriteDebugInfo() {
+// TODO this launcher should be generalized, maybe stuck as a general 'c10'
+// helper
```

Comment:
@zdevito i was thinking of leaving this for a follow up PR.  also do you have a suggestion where to put the util and what its interface should be?

i didn't do it here mainly to avoid figuring out the varargs and captures needed to pass args to the inner function from outside the launcher utility, but those should be solve-able with enough chat-gpting.